modules/fedimint-mint-client/src/lib.rs

```diff
@@ -909,6 +909,29 @@ impl ClientModule for MintClientModule {
                 }),
         )
     }
+
+    async fn leave(
+        &self,
+        dbtx: &mut ModuleDatabaseTransaction<'_>,
+        module_instance_id: ModuleInstanceId,
+        executor: Executor<DynGlobalClientContext>,
+        _api: DynGlobalApi,
+    ) -> anyhow::Result<()> {
+        let balance = ClientModule::get_balance(self, dbtx).await;
+        if Amount::from_sats(0) < balance {
+            bail!("Outstanding balance: {balance}");
+        }
+
+        if executor
+            .get_active_states()
+            .await
+            .into_iter()
+            .any(|s| s.0.module_instance_id() == module_instance_id)
+        {
+            bail!("Pending operations")
+        }
```

Comment:
This code looks to be module-agnostic. Can we provide it to modules for free without having to write it by hand?

Reply:
Possibly sometehing to do along https://github.com/fedimint/fedimint/issues/3447